internal/kubernetes/operator/deployment/deployment.go

```diff
@@ -171,9 +171,36 @@ func BuildAtlasAdvancedDeployment(deploymentStore store.OperatorClusterStore, va
 		advancedSpec.ManagedNamespaces = managedNamespaces
 	}
 
+	cleanTenantFields(&atlasDeployment.Spec)
+
 	return deploymentResult, nil
 }
 
+func cleanTenantFields(out *akov2.AtlasDeploymentSpec) {
+	isTenant := false
+
+	if out.DeploymentSpec == nil {
+		return
+	}
+	for _, spec := range out.DeploymentSpec.ReplicationSpecs {
+		for _, c := range spec.RegionConfigs {
+			if c.ProviderName == "TENANT" {
+				isTenant = true
+				break
+			}
+		}
+	}
+
+	if isTenant {
+		out.DeploymentSpec.BiConnector = nil
+		out.DeploymentSpec.EncryptionAtRestProvider = ""
+		out.DeploymentSpec.DiskSizeGB = nil
+		out.DeploymentSpec.MongoDBMajorVersion = ""
+		out.DeploymentSpec.PitEnabled = nil
+		out.DeploymentSpec.BackupEnabled = nil
+	}
+}
```

Comment:
This misses a couple of nil checks and can panic. Suggestion:

```
...
	if hasTenantFields(&atlasDeployment.Spec) {
		atlasDeployment.Spec.DeploymentSpec.BiConnector = nil
		atlasDeployment.Spec.DeploymentSpec.EncryptionAtRestProvider = ""
		atlasDeployment.Spec.DeploymentSpec.DiskSizeGB = nil
		atlasDeployment.Spec.DeploymentSpec.MongoDBMajorVersion = ""
		atlasDeployment.Spec.DeploymentSpec.PitEnabled = nil
		atlasDeployment.Spec.DeploymentSpec.BackupEnabled = nil
	}

	return deploymentResult, nil
}

func hasTenantFields(out *akov2.AtlasDeploymentSpec) bool {
	if out.DeploymentSpec == nil {
		return false
	}
	for _, spec := range out.DeploymentSpec.ReplicationSpecs {
		if spec == nil {
			continue
		}
		for _, c := range spec.RegionConfigs {
			if c == nil {
				continue
			}
			if c.ProviderName == "TENANT" {
				return true
			}
		}
	}
	return false
}
```

Reply:
How can `spec` or `c` be nil if we're ranging over them? If there's nothing there, we won't enter the loop, right?

Reply:
@roothorp the following two cases panic with this code:
```
	spec := akov2.AtlasDeploymentSpec{
		DeploymentSpec: &akov2.AdvancedDeploymentSpec{
			ReplicationSpecs: []*akov2.AdvancedReplicationSpec{
				{
					nil,
				},
			},
		},
	}

	cleanTenantFields(&spec)
```

Reply:
```
	spec := akov2.AtlasDeploymentSpec{
		DeploymentSpec: &akov2.AdvancedDeploymentSpec{
			ReplicationSpecs: []*akov2.AdvancedReplicationSpec{
				{
					RegionConfigs: []*akov2.AdvancedRegionConfig{
						nil,
					},
				},
			},
		},
	}

	cleanTenantFields(&spec)
```

Reply:
done!